src/Angor/Client/Pages/Spend.razor

```diff
@@ -880,34 +870,57 @@
     {
         var selectedItem = e.Value?.ToString();
 
-        if (selectedItem != null)
+        if (selectedItem != null && int.TryParse(selectedItem, out int res) && res <= feeData.FeeEstimations.Fees.Count)
         {
-            if (int.TryParse(selectedItem, out int res))
-            {
-                if (res <= feeData.FeeEstimations.Fees.Count)
+            feeData.FeePosition = res;
+
+            var selectedFeeEstimation = feeData.FeeEstimations.Fees.OrderBy(fee => fee.Confirmations).ToList()[res - 1];
+
+            await RecalculateFee(selectedFeeEstimation.FeeRate, selectedFeeEstimation.Confirmations);
+        }
+    }
+
+    private async Task RecalculateFee(decimal customFeeRate, int confirmations = 0)
```

Comment:
why decimal isn't the fee represented as a long?

Reply:
It was a mistake, and I’ve fixed it. Thanks for pointing it out!